src/crimson/net/SocketMessenger.cc

```diff
@@ -23,9 +23,22 @@
 using namespace ceph::net;
 
 SocketMessenger::SocketMessenger(const entity_name_t& myname,
-                                 const std::string& logic_name)
-  : Messenger{myname}, logic_name{logic_name}
-{}
+                                 const std::string& logic_name,
+                                 uint64_t nonce)
+  : Messenger{myname}, logic_name{logic_name}, nonce{nonce}
+{
+  entity_addr_t my_addr;
+  my_addr.set_type(entity_addr_t::TYPE_DEFAULT);
```

Comment:
why do we need an empty `TYPE_LEGACY` here? IIUC, messenger will learn its address or will have it when `set_myaddr()` (or `set_myaddrs()` in v2) gets called.

Reply:
The reason:
https://github.com/ceph/ceph/blob/master/src/msg/msg_types.cc#L172-L174
An entity_addr_t with TYPE_NONE will always print "-", even though it may have a meaningful nonce value.

What do you think of this way: modify `operator<<()` of `entity_addr_t` to print out its nonce unconditionally?

Reply:
> What do you think of this way: modify `operator<<()` of `entity_addr_t` to print out its nonce unconditionally?

i see. but i think this change actually defeats the purpose to avoid printing the address if it is not initialized, i.e., it's empty.

because the nonce of an empty `entity_addr_t` is always the same, as i think it's just the pid of current process, what the point of having it in the log message?

Reply:
also this change is not related to the fix. could you split it into its own commit?

Reply:
> as i think it's just the pid of current process, what the point of having it in the log message?

The logic here is to print out `nonce` of the messenger before the address binded or learned, as `nonce` is an important part (works like pid) to uniquely identify an messenger. What do you think of this? I'm also ok to print "-" in this case.

Reply:
without the `logic_name`, we can hardly identify a messenger merely using its nonce. instead, i think, the tuple of <`type`, `nonce`> is used to identify an entity in ceph cluster. for instance, "osd.123".

 and i don't quite understand why we need to track a not-yet-connected messenger.

Reply:
removed.